### PR TITLE
release version v1.4.0

### PR DESCRIPTION
**Pull request by 3062**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

**Comment by nejisama** [comment]
需要更新 mosn.io/pkg ，并且需要记录changelog

File: CHANGELOG_ZH.md [hunk 1]
```diff
@@ -1,5 +1,38 @@
...
 
+## v1.4.0
+
+### 新功能
+
+- 支持记录 HTTP 健康检查日志 (#2096) @dengqian
+- 新增最小连接负载均衡器 (#2184) @dengqian
+- 新增 API: 强制断开并重新连接 ADS 服务 (#2183) @dengqian
+- 支持 pprof debug server 配置 endpoint (#2202) @dengqian
+- 集成 mosn.io/envoy-go-extension (#2200) @antJack (#2222) @3062
```
Comment (doujiang24) [inline CHANGELOG_ZH.md:11]: https://github.com/mosn/mosn/blob/master/examples/codes/envoy-go-extension/README_CN.md
可以加一下这个示例文档

File: CHANGELOG_ZH.md [hunk 2]
```diff
@@ -1,5 +1,38 @@
...
+- 新增 API: 重写 Variable (mosn/pkg#72) @antJack
```
Comment (doujiang24) [inline CHANGELOG_ZH.md:12]:
```suggestion
- 新增 API: 支持覆盖注册 Variable (mosn/pkg#72) @antJack
```

File: pkg/types/healthcheck.go [hunk 3]
```diff
@@ -29,7 +29,6 @@ const (
 
 // HealthCheckCb is the health check's callback function
 type HealthCheckCb func(host Host, changedState bool, isHealthy bool)
-// HealthCheckCb is the health check's callback function
```
Comment (doujiang24) [inline pkg/types/healthcheck.go:32]: 这些非 release 相关的改动，单独搞一个 PR，更合适一些

File: CHANGELOG_ZH.md (hunk 1 above)
Comment (3062) [reply CHANGELOG_ZH.md]: 新的提交再看一下呢？我不确定是不是要这么写

Comment (doujiang24) [reply CHANGELOG_ZH.md]: 地址要搞成可以点击的链接，[example](https://github.com/mosn/mosn/blob/master/examples/codes/envoy-go-extension/README_CN.md)

以前的 release 日志，可以参考
https://github.com/mosn/mosn/blob/master/CHANGELOG_ZH.md#v110

**Review by doujiang24 — DISMISSED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,38 @@
...
+- Support pprof debug server endpoint configure (#2202) @dengqian
+- Integrate with mosn.io/envoy-go-extension, consult [document](https://github.com/mosn/mosn/blob/master/examples/codes/envoy-go-extension/README_EN.md) (#2200) @antJack (#2222) @3062
+- Add API to support override registration (mosn/pkg#72) @antJack
+
+### Refactoring
+- Move pprof import to pkg/mosn (#2216) @3062
+
+### Optimization
+
+- Reduces logging for proxywasm benchmarks (#2189) @Crypt Keeper
```
Comment (taoyuanyuan) [inline CHANGELOG.md:19]: Reduce

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,38 @@
...
+- Fix where new mosn would delete reconfig.sock when smooth upgrade failed (#2205) @dengqian
+- Fix HTTP health check URL with query string will be unescaped (#2207) @dengqian
+- Fix lb ReqRoundRobin fails to choose host when index exceeds hosts length (#2209) @dengqian
+- Fix the problem that after RDS creates a route, the established connection cannot find the route (#2199) @dengqian (#2210) @3062
+- Fix old cache values are read after the execution of Variable.Set (mosn/pkg#73) @antJack
+- Fix panic caused by DefaultRoller not setting time (mosn/pkg#74) @dengqian
+- Advance the timing of metrics initialization to make it work for static config (#2221) @dengqian
+- Fix a concurrency problem caused by multiple health checkers sharing rander (#2228) @dengqian
+- Set HTTP1.1 as the HTTP protocol to be sent upstream (#2225) @dengqian
+- Add a field in variables to record mosn processing time (#2235) @z2z23n0
```
Comment (taoyuanyuan) [inline CHANGELOG.md:34]: It's a feature.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,38 @@
...
+- Fix the debug server not inherited during smooth upgrade (#2204) @dengqian
+- Fix where new mosn would delete reconfig.sock when smooth upgrade failed (#2205) @dengqian
+- Fix HTTP health check URL with query string will be unescaped (#2207) @dengqian
+- Fix lb ReqRoundRobin fails to choose host when index exceeds hosts length (#2209) @dengqian
+- Fix the problem that after RDS creates a route, the established connection cannot find the route (#2199) @dengqian (#2210) @3062
+- Fix old cache values are read after the execution of Variable.Set (mosn/pkg#73) @antJack
+- Fix panic caused by DefaultRoller not setting time (mosn/pkg#74) @dengqian
+- Advance the timing of metrics initialization to make it work for static config (#2221) @dengqian
+- Fix a concurrency problem caused by multiple health checkers sharing rander (#2228) @dengqian
+- Set HTTP1.1 as the HTTP protocol to be sent upstream (#2225) @dengqian
```
Comment (taoyuanyuan) [inline CHANGELOG.md:33]: HTTP/1.1

File: CHANGELOG_ZH.md [hunk 7]
```diff
@@ -1,5 +1,38 @@
...
+
+### 新功能
+
+- 支持记录 HTTP 健康检查日志 (#2096) @dengqian
+- 新增最小连接负载均衡器 (#2184) @dengqian
+- 新增 API: 强制断开并重新连接 ADS 服务 (#2183) @dengqian
+- 支持 pprof debug server 配置 endpoint (#2202) @dengqian
+- 集成 mosn.io/envoy-go-extension，参考[文档](https://github.com/mosn/mosn/blob/master/examples/codes/envoy-go-extension/README_CN.md) (#2200) @antJack (#2222) @3062
+- 新增 API: 支持覆盖注册 Variable (mosn/pkg#72) @antJack
+- 在 variables 中增加记录 mosn 处理时间的字段 (#2235) @z2z23n0
```
Comment (taoyuanyuan) [inline CHANGELOG_ZH.md:13]: 新增记录 mosn 处理时间的字段的变量

File: CHANGELOG_ZH.md [hunk 8]
```diff
@@ -1,5 +1,38 @@
...
+
+### 重构
+- pprof 导入迁移至 pkg/mosn (#2216) @3062
```
Comment (taoyuanyuan) [inline CHANGELOG_ZH.md:16]: 导入 ？

File: CHANGELOG_ZH.md [hunk 9]
```diff
@@ -1,5 +1,38 @@
...
+- 修复平滑升级时未继承 debug server 的问题 (#2204) @dengqian
+- 修复平滑升级失败时，新 mosn 会删除 reconfig.sock 的问题 (#2205) @dengqian
+- 修复 HTTP 健康检查 URL query string 未转译的问题 (#2207) @dengqian
+- 修复 ReqRoundRobin 负载均衡器在索引超过 host 数量时，host 选择失败的问题 (#2209) @dengqian
+- 修复 RDS 创建路由之后，已建连的连接无法找到路由的问题 (#2199) @dengqian (#2210) @3062
+- 修复 Variable.Set 执行后会读取到旧缓存值的问题 (mosn/pkg#73) @antJack
+- 修复 DefaultRoller 未设置时间导致 panic 的问题 (mosn/pkg#74) @dengqian
+- 提前 metrics 初始化时间使其适用于 static config (#2221) @dengqian
+- 修复多个 health checker 共用 rander 导致的并发问题 (#2228) @dengqian
+- 设置 HTTP1.1 作为发往上游的 HTTP 协议 (#2225) @dengqian
```
Comment (taoyuanyuan) [inline CHANGELOG_ZH.md:34]: HTTP/1.1

File: CHANGELOG_ZH.md (hunk 8 above)
Comment (3062) [reply CHANGELOG_ZH.md]: 改成了 import pprof

**Review by doujiang24 — APPROVED** [review]

**Review by antJack — APPROVED** [review]
LGTM

**Pull request merged**